tensorflow-framework/src/main/java/org/tensorflow/framework/data/Dataset.java

```diff
@@ -107,6 +123,72 @@ public final Dataset take(long count) {
         tf, this.getVariant(), tf.constant(count), this.getOutputTypes(), this.getOutputShapes());
   }
 
+  /**
+   * Returns a new Dataset which maps a function across all elements from this iterator, on a single
+   * component of each element.
+   *
+   * <p>For example, suppose each element is a `List<Operand<?>>` with 2 components: (features,
+   * labels).
+   *
+   * <p>Calling `iterator.mapOneComponent(0, features -> tf.math.mul(features, tf.constant(2)))`
+   * will map the function over the `features` component of each element, multiplying each by 2.
+   *
+   * @param index The index of the component to transform.
+   * @param mapper The function to apply to the target component.
+   * @return A new DatasetIterator applying `mapper` to the component at the chosen index.
+   */
+  public Dataset mapOneComponent(int index, Function<Operand<?>, Operand<?>> mapper) {
+    return map(
+        outputs -> {
+          List<Operand<?>> newComponents = new ArrayList<>(outputs);
+          newComponents.set(index, mapper.apply(outputs.get(index)));
+          return newComponents;
+        });
+  }
+
+  /**
+   * Returns a new Dataset which maps a function across all elements from this iterator, on all
+   * components of each element.
+   *
+   * <p>For example, suppose each element is a `List<Operand<?>>` with 2 components: (features,
+   * labels).
+   *
+   * <p>Calling `iterator.mapAllComponents(component -> tf.math.mul(component, tf.constant(2)))`
+   * will map the function over the both the `features` and `labels` components of each element,
+   * multiplying them all by 2
+   *
+   * @param mapper The function to apply to each component
+   * @return A new DatasetIterator applying `mapper` to all components of each element.
+   */
+  public Dataset mapAllComponents(Function<Operand<?>, Operand<?>> mapper) {
+    return map(
+        outputs -> {
+          List<Operand<?>> mappedOutputs = new ArrayList<>();
+          outputs.forEach(o -> mappedOutputs.add(mapper.apply(o)));
+          return mappedOutputs;
+        });
+  }
+
+  /**
+   * Returns a new Dataset which maps a function over all elements returned by this iterator.
+   *
+   * <p>For example, suppose each element is a `List<Operand<?>>` with 2 components: (features,
+   * labels).
+   *
+   * <p>Calling ``` iterator.map(components -> { Operand<?> features = components.get(0); Operand<?>
+   * labels = components.get(1);
+   *
+   * <p>return Arrays.asList( tf.math.mul(features, tf.constant(2)), tf.math.mul(labels,
+   * tf.constant(5)) ); }) ``` will map the function over the `features` and `labels` components,

```

Comment:
again instead of triple backquotes ``` (which should help with Javadoc, did it?, you should go with a block like this:
```java
<pre>{@code
your 
multiline 
code
here
}</pre>
```

Reply:
Thanks, updated!